registry_provider_version.go

```diff
@@ -0,0 +1,262 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+)
+
+// Compile-time proof of interface implementation.
+var _ RegistryProviderVersions = (*registryProviderVersions)(nil)
+
+// RegistryProviderVersions describes the registry provider version methods that
+// the Terraform Enterprise API supports.
+//
+// TFE API docs: https://www.terraform.io/cloud-docs/api-docs/private-registry/provider-versions-platforms
+type RegistryProviderVersions interface {
+	// List all versions for a single provider.
+	List(ctx context.Context, providerID RegistryProviderID, options *RegistryProviderVersionListOptions) (*RegistryProviderVersionList, error)
+
+	// Create a registry provider version.
+	Create(ctx context.Context, providerID RegistryProviderID, options RegistryProviderVersionCreateOptions) (*RegistryProviderVersion, error)
+
+	// Read a registry provider version.
+	Read(ctx context.Context, versionID RegistryProviderVersionID) (*RegistryProviderVersion, error)
+
+	// Delete a registry provider version.
+	Delete(ctx context.Context, versionID RegistryProviderVersionID) error
+}
+
+// registryProvidersVersions implements RegistryProvidersVersions
+type registryProviderVersions struct {
+	client *Client
+}
+
+// RegistryProviderVersion represents a registry provider version
+type RegistryProviderVersion struct {
+	ID                 string                      `jsonapi:"primary,registry-provider-versions"`
+	Version            string                      `jsonapi:"attr,version"`
+	CreatedAt          string                      `jsonapi:"attr,created-at,iso8601"`
+	UpdatedAt          string                      `jsonapi:"attr,updated-at,iso8601"`
+	KeyID              string                      `jsonapi:"attr,key-id"`
+	Protocols          []string                    `jsonapi:"attr,protocols,omitempty"`
+	Permissions        RegistryProviderPermissions `jsonapi:"attr,permissions"`
+	ShasumsUploaded    bool                        `jsonapi:"attr,shasums-uploaded"`
+	ShasumsSigUploaded bool                        `jsonapi:"attr,sasums-sig-uploaded"`
+
+	// Relations
+	RegistryProvider          *RegistryProvider           `jsonapi:"relation,registry-provider"`
+	RegistryProviderPlatforms []*RegistryProviderPlatform `jsonapi:"relation,platforms"`
+
+	// Links
+	Links map[string]interface{} `jsonapi:"links,omitempty"`
+}
+
+// RegistryProviderVersionID is the multi key ID for addressing a version provider
+type RegistryProviderVersionID struct {
+	RegistryProviderID
+	Version string
+}
+
+type RegistryProviderVersionList struct {
+	*Pagination
+	Items []*RegistryProviderVersion
+}
+
+type RegistryProviderVersionListOptions struct {
+	ListOptions
+}
+
+type RegistryProviderVersionCreateOptions struct {
+	Version   string   `jsonapi:"attr,version"`
+	KeyID     string   `jsonapi:"attr,key-id"`
+	Protocols []string `jsonapi:"attr,protocols"`
```

Comment:
Suggestion to make this strongly typed (see this [comment](https://github.com/hashicorp/go-tfe/pull/405/files#r876436790)).

Reply:
Per an internal discussion, I'm going to hold off on making the protocols strongly typed because there is a discrepancy between internal documents regarding what the values should be. I've opened a ticket to look at this issue more.

Reply:
Once we release this, we can't really change it without incrementing the major version since its part of the public interface. I think we should decide now if this is the API we want and then never change it.

Reply:
My 2 🪙  if the `Protocols` field is a limited set of known values, they should definitely be enumerated. This is a pattern we've enforced with v1.0. 
```golang
type ProviderProtocol string
const (
   // values here
)
```

Reply:
@brandonc and I chatted about this field today. and it seems like the intent is to provide major version compatibility. Given that this is a library, it seems like our best bet is to keep this field a string, and let the API do the checking (which it does currently).